scripts/linux_containers_on_windows/build_linux_kernel.sh

```diff
@@ -0,0 +1,172 @@
+#!/bin/bash
+
+set -xe -o pipefail
+
+source ~/.profile
+
+LINUX_BASE_DIR=""
+OPENGCS_BASE_DIR=""
+
+function get_job_number (){
+    #
+    # Multiply current number of threads with a number
+    # Usage:
+    #   ./build_artifacts.sh --thread_number x10
+    #
+    multi="${1#*x}"
+    cores="$(cat /proc/cpuinfo | grep -c processor)"
+    result="$(expr $cores*$multi | bc)"
+    echo ${result%.*}
+}
+
+function apply_patches() {
+    #
+    # Apply some patches before building
+    #
+    linux_base_dir="$1"; shift
+    linux_msft_artifacts_destination_path="$1"; shift
+    kernel_version="$1"; shift
+    checkout_commit="$1"; shift
+    cherry_pick_commit="$1"; shift
+    opengcs_base_dir="$1"; shift
+    commits_list="$1"
+
+
+    if [[ ! -d "${linux_base_dir}/msft_linux_kernel" ]]; then
+        echo "Could not find the Linux Kernel repo to apply patches"
+        exit 1
+    else
+        pushd "${linux_base_dir}/msft_linux_kernel"
+    fi
+
+    # this will be the destination directory of the kernel artifact
+    output_dir_name="${linux_msft_artifacts_destination_path}/$(date +%Y%m%d)_${BUILD_ID}__msft-kernel_${kernel_version}"
+    mkdir -p "${output_dir_name}"
+
+    # last commit before pulling
+    echo $(git log --pretty=format:'%h' -n 1) > "${output_dir_name}/latest_kernel_commit.log"
+    echo "Linux kernel built on commit:"
+    cat "${output_dir_name}/latest_kernel_commit.log"
+
+    # tag for specific repo
+    branch="$(git rev-parse --abbrev-ref HEAD)"
+    if [[ "${branch}" == "master" ]]; then
+        git checkout "${checkout_commit}"
+        git reset --hard HEAD~20 #Note (papagalu): if the env is not clean, we have to revert all cherry-picks
+        git cherry-pick "$cherry_pick_commit"
+
+        echo "Apply NVDIMM patch"
+        patch -p1 -t <"${opengcs_base_dir}/kernel/patches-4.12.x/0002-NVDIMM-reducded-ND_MIN_NAMESPACE_SIZE-from-4MB-to-4K.patch"
+        cp "${opengcs_base_dir}/kernel/kernel_config-${kernel_version}.x" .config
+
+        echo "Instructions for getting Hyper-V vsock patch"
+        git remote add -f dexuan-github https://github.com/dcui/linux.git  || echo "already existing"
+        git fetch dexuan-github
+
+        IFS=$','
+        for commit in ${COMMITS_LIST[@]}; do
+            git cherry-pick -x "${commit}"
+        done
+        IFS=$' '
+
+        popd
+        echo "Patches applied on Linux Kernel successfully"
+    fi
+}
+
+function build_kernel() {
+    linux_base_dir="$1"; shift
+    linux_msft_artifacts_destination_path="$1"; shift
+    kernel_version="$1"; shift
+    thread_num="$1"
+
+    pushd "${linux_base_dir}/msft_linux_kernel"
+
+    echo "Building the LCOW MS-Linux kernel"
+    fakeroot make -j"${thread_num}" && sudo make modules
```

Comment:
do you need the sudo? can't you do
fakeroot make -j"${thread_num}" && fakeroot make modules